packages/amplify-graphql-schema-generator/src/datasource-adapter/pg-string-datasource-adapter.ts

```diff
@@ -118,7 +118,9 @@ export class PostgresStringDataSourceAdapter extends StringDataSourceAdapter {
   }
 
   public getPrimaryKey(tableName: string): Index | null {
-    const key = this.indexes.find((index) => index.tableName === tableName && index.indexName === `${tableName}_pkey`);
+    const key = this.indexes.find(
+      (index) => index.tableName === tableName && index.indexName.match(/.*_pkey/) && index.columns.includes(index.columnName),
```

Comment:
This is still using name matching on index to determine PK. If we have indices named as `my_index_pk` we would would mistake them to be primaryKey constraints?

Reply:
I believe Postgres would throw an error here when trying to create the table. In our `news` test schema if I change `CONSTRAINT publications_superkey UNIQUE (pub_id, pub_type)` to `CONSTRAINT publications_pkey UNIQUE (pub_id, pub_type)` the query fails with: `ERROR:  relation "publications_pkey" already exists`.